File: solo-leveling-user-service/src/main/java/com/sleepkqq/sololeveling/user/service/config/GrpcConfig.java
package com.sleepkqq.sololeveling.user.service.config;

import io.envoyproxy.pgv.ValidatorImpl;
import org.springframework.beans.factory.annotation.Configurable;
import org.springframework.context.annotation.Bean;

@Configurable
public class GrpcConfig {

  @Bean
  ValidatorImpl validator() {

  }

}

File: solo-leveling-player-service/src/main/kotlin/com/sleepkqq/sololeveling/player/service/config/GrpcConfig.kt
package com.sleepkqq.sololeveling.player.service.config

import com.sleepkqq.sololeveling.proto.config.LocaleServerInterceptor
import io.grpc.ServerInterceptor
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.core.annotation.Order
import org.springframework.grpc.server.GlobalServerInterceptor

@Suppress("unused")
@Configuration
class GrpcConfig {

	@Bean
	@Order(100)
	@GlobalServerInterceptor
	fun localeServerInterceptor(): ServerInterceptor = LocaleServerInterceptor()
}
